### PR TITLE
Fixes #30348 - trends cleanup task

### PR DESCRIPTION
**Pull request by ezr-ondrej**
Cleans up trends data for users who wish to stop using trends.

### CODE REVIEW

File: lib/tasks/trends.rake [hunk 1]
```diff
@@ -1,4 +1,16 @@
...
+  task :clobber => :environment do
+    class FakeTrend < ApplicationRecord
+      self.table_name = 'trends'
+    end
+    class FakeTrendCounter < ApplicationRecord
+      self.table_name = 'trend_counters'
+    end
+    FakeTrendCounter.delete_all
+    FakeTrend.delete_all
+  end
```
Comment (tbrisker) [inline lib/tasks/trends.rake]: We need to also cleanup the db tables, permissions and settings when running this task. Should we put it in a different namespace where we can put additional cleanup tasks in the future?

File: lib/tasks/trends.rake [hunk 2]
```diff
@@ -1,4 +1,16 @@
 namespace :trends do
+  desc 'Clean up Trends data'
+  task :clobber => :environment do
```
Comment (ekohl) [inline lib/tasks/trends.rake]: `clobber` doesn't tell me exactly what it'd do. How about `purge`?

File: lib/tasks/trends.rake (hunk 1 above)
Comment (ezr-ondrej) [reply lib/tasks/trends.rake]: My :heart: hurts handling schema anywhere else then migrations. Do we really want to drop the tables from the task? Given there should be a migration that will drop all the data for everyone.
Setting and perms I agree :100:  :+1:

**Comment by ekohl** [comment]
There is still a cronjob:

https://github.com/theforeman/foreman-packaging/blob/56cebd1c4617d67b0e7a9f33fb5541f0b4b1e50e/packages/foreman/foreman/foreman.cron.d#L18-L19
https://github.com/theforeman/foreman-packaging/blob/caa5b6b984d60da8f9eee2b77a4e589376b3e2e4/debian/bionic/foreman/foreman.cron.d#L18-L19
https://github.com/theforeman/foreman-packaging/blob/caa5b6b984d60da8f9eee2b77a4e589376b3e2e4/debian/buster/foreman/foreman.cron.d#L18-L19

With that present, it'll quickly be populated again. Don't forget to clean that up as well.

File: lib/tasks/trends.rake (hunk 1 above)
Comment (tbrisker) [reply lib/tasks/trends.rake]: I'm not sure about dropping it for everyone in migration, what if i upgraded and didn't mean to remove the data but only realized that i need to install the plugin too late?

**Comment by tbrisker** [comment]
@ezr-ondrej those cron jobs probably need to move in packaging to the plugin's package

File: lib/tasks/trends.rake (hunk 1 above)
Comment (ezr-ondrej) [reply lib/tasks/trends.rake]: well that's a bummer and tough life but that is what it is... You've had two versions to realize that.
I mean assuming we are going to add the migration with droping the deprecated endpoints.

**Comment by ezr-ondrej** [comment]
I've moved the cron jobs to a Dynflow action: https://github.com/theforeman/foreman_statistics/pull/19
I've moved the task to the namespace `purge` and introduced new rake `purge_data`, that can be used for future purges and that should cleanup all unnecessary data.

File: lib/tasks/trends.rake (hunk 1 above)
Comment (tbrisker) [reply lib/tasks/trends.rake]: I think it makes sense to drop here as well - when someone runs this task, they are saying "we don't want this stuff, get rid of it", which to me means we should also drop the tables. We can still add a migration as well to make sure we remove them when we drop the api endpoints for those who were lazy and didn't care about running the task.

**Comment by ezr-ondrej** [comment]
I've deleted the tables. Just a note to consider, the plugin is not installable after run of this script in current setting.
Should we squash migrations in plugin then?

**Comment by ezr-ondrej** [comment]
@tbrisker do you have more comments?

**Comment by tbrisker** [comment]
@ezr-ondrej could we somehow also delete the migrations from the db? that way if the plugin is installed after this task is run it will think those migrations never executed.
Also, still need to clean up permissions and settings.

**Comment by ezr-ondrej** [comment]
Rake is dropping Permission, Setting and removes records in the SchemaMigration now.

**Review by tbrisker — APPROVED** [review]
Thanks @ezr-ondrej !

**Review by tbrisker — APPROVED** [review]
tested, dropping works and so does re-migrating afterwards with the plugin installed.

**Comment by tbrisker** [comment]
2.2 - b01ab75b2

**Comment by tbrisker** [comment]
I just realized there's also permissions for statistics that we should clean up - could you please add a follow up pr @ezr-ondrej ?

**Comment by ezr-ondrej** [comment]
Sure: #7932

**Pull request merged**